util/progress/printer.go

```diff
@@ -34,15 +41,25 @@ func NewPrinter(ctx context.Context, out console.File, mode string) *Printer {
 		done:   doneCh,
 	}
 
-	if v := os.Getenv("BUILDKIT_PROGRESS"); v != "" && mode == "auto" {
+	if v := os.Getenv("BUILDKIT_PROGRESS"); v != "" && mode == PrinterModeAuto {
 		mode = v
 	}
 
 	go func() {
 		var c console.Console
-		if cons, err := console.ConsoleFromFile(out); err == nil && (mode == "auto" || mode == "tty") {
-			c = cons
+		switch mode {
+		case PrinterModeQuiet:
+			devNull, err := os.Open(os.DevNull)
```

Comment:
just set out to `ioutil.Discard` , no `fallthrough` needed

Reply:
`ioutil.Discard` is a `Writer` and `out` is a `File`.

About the `fallthrough` I'm not sure if I understood, but it is needed in order to execute `console.ConsoleFromFile(out)` and assign to `c`. Otherwise, how could it affect the console? Note that if not, `plain` output is printed.

Reply:
The parameter that `DisplaySolveStatus` takes is an `io.Writer`. If you discard the output then you don't have a console. That should answer to the `fallthrough` case as well.